### ISSUE TITLE
AndySky21 PRs

### ISSUE THREAD

**Issue by AndySky21** [doublespeakgames/adarkroom#397]
@Continuities 
I see that I posted a lot of PRs, it's way easier for me to post than for you to check, so here is a list in order of increasing complexity.

Useful and easy
~~#391 a very small bugfix. Browser title is not translated on Ship panel.~~
~~#385 It translation: "save." is translated with a noun instead of a verb (fits better with the Import/Export command)~~
~~#396 (if you like it) enemy letters (editorial change)~~
~~#378 notification for outside events. A bunch of advices for missing lines (not sure if they fit into the atmosphere. Needs the hand of game Creator)~~

Quite useful and easy
~~#411 bugfix for overlapping up-down buttons, so they're easier to click~~
~~#390 ordered lists for workers and travel supplies~~
~~#380 bugfix. this disables tab navigation while in the wild, and re-enables it when you come back or die. Prevents horizontal return movements to also trigger panel change.~~

Improvements that could need some time for reviewing
~~#399 is quite urgent, because current take-all system is buggy. This adds new features and fixes the bug.~~
~~#386 a small logical fix. Clicking "got it" will close export panel, while aborting import will open Import/Export selection (maybe the player wants to save current game before trying a reboot).~~
~~#360 it's (in)famous and @grotewold was reviewing it. Compass direction tooltip until W is found. (Mind that a new PR indirectly depends on it. Lol.)~~
#375 button cooldown stored in state.
~~#372 mysterious wanderers return when reloading if they were "good".~~

Less necessary improvements
~~#379 a function for randomly destroy one or more huts. If it kills the last, it could have less than 4 inhabitants. Could also be used to destroy a hut without necessarily killing people.~~
~~#389 a translation system for box "legends" which uses CSS and does not need an additional inline stylesheet. Translated legends are stored as attributes.~~

Edit: the following are not PRs yet, as it would be really too much of a nuisance as of now. But their code is ready.
- ~~a new "Take all and leave" approach, with buttons to gather "all of a type". Removed leave cooldown.~~
- fight panel buttons reordering: attack buttons, leave buttons and heal buttons are divided into specific groups so that they don't get confused.
- Don't heal the healthy: eat meat & use meds buttons are disabled unless you lose some health (and they get disabled when cured). Prevents wasting of resources.
- Pause button: a "pause."/"resume" button for fights. It disables attack and heal actions (without disabling buttons themselves), temporarily freezes cooldown and stops enemy attacks.

**Comment by AndySky21** [comment]
Note: if you merge "compass tooltip", I have a "production store split & productivity flow show" ready to PR.
On the other hand, if you merge my last "Take everything possible", I'll send you a nice little fix that separates scene buttons (e.g. continue and leave) from heal buttons (Eat meat and Use meds), but, most important, it blocks/unblocks Meat/Meds buttons depending on whether you actually need to be cured. A little help for players, a visual indication if they're OK or no.
I can't do it now, because I don't branch in my own copy, so changes are all together.
Punch me in da face.

**Comment by AndySky21** [comment]
Updated 2015-06-18. Let me know if there are issues with any commit.
See you soon.

**Comment by skj3gg** [comment]
Thanks. I'll make it a goal to finish three of these beauties today

**Comment by grotewold** [comment]
2 left!  ....until new PRs :frowning: :smile:

**Comment by AndySky21** [comment]
I celebrated my birthday some days ago. But seeing THIS was a far better celebration! Thank you for the effort in keeping ADR up-to-date with user comments, and for the satisfaction you gave me!
I'll fix #360 and #375 ASAP.

**Comment by grotewold** [comment]
YEAHHH!!!!!!!!!!!!!!!!!!111!!1!!!!!!!!!!!!!!!11111!!!!!!!!!!!!!

**Comment by AndySky21** [comment]
When I'm sad, I'll just re-read all your answers to my past PRs :)
ADR is the second Github project I contribute to, but the first relevant (the other one was a Mediawiki extension).
The only thing I enjojed more than contributing to the project, was to play the game itself. I'm proud to have contributed somehow to this fantastic game, and you gave me a lot of satisfaction with all of my PRs, thank you so much!
